### PR TITLE
[cloud] Fix repo syncer for organization repositories

### PR DESCRIPTION
**Pull request by kopancek**
# Description

We didn't apply any creation limits
We didn't track org owners for these repos properly

Both should be fixed by this commit

# Related Jira issue
https://sourcegraph.atlassian.net/browse/CLOUD-164

# Testing

* Modified unit tests to verify the changes to limits
* Not sure how to change the metric collection

### CODE REVIEW

File: internal/repos/store.go [hunk 1]
```diff
@@ -138,38 +137,39 @@ func (s *Store) trace(ctx context.Context, family string) (*trace.Trace, context
 	return tr, ctx
 }
 
-// CountUserAddedRepos counts the total number of repos that have been added
+// CountNamespacedRepos counts the total number of repos that have been added
 // by user owned external services. If userIDs are specified, only repos owned by the given
 // users are counted.
-func (s *Store) CountUserAddedRepos(ctx context.Context, userIDs ...int32) (count uint64, err error) {
-	tr, ctx := s.trace(ctx, "Store.CountUserAddedRepos")
+func (s *Store) CountNamespacedRepos(ctx context.Context, userID int32, orgID int32) (count uint64, err error) {
```
Comment (unknwon) [inline internal/repos/store.go:143]:
```suggestion
func (s *Store) CountNamespacedRepos(ctx context.Context, userID, orgID int32) (count uint64, err error) {
```

File: internal/repos/syncer.go [hunk 2]
```diff
@@ -436,6 +457,7 @@ func (s *Syncer) SyncExternalService(
 	// Unless our site config explicitly allows private code or the user has the
 	// "AllowUserExternalServicePrivate" tag, user added external services should
 	// only sync public code.
+	// Organization owned external services are always considered allowed
```
Comment (unknwon) [inline internal/repos/syncer.go:460]:
```suggestion
	// Organization owned external services are always considered allowed.
```

**Review by unknwon — APPROVED** [review]

File: internal/repos/store.go [hunk 3]
```diff
@@ -138,38 +137,39 @@ func (s *Store) trace(ctx context.Context, family string) (*trace.Trace, context
 	return tr, ctx
 }
 
-// CountUserAddedRepos counts the total number of repos that have been added
+// CountNamespacedRepos counts the total number of repos that have been added
 // by user owned external services. If userIDs are specified, only repos owned by the given
```
Comment (ryanslade) [inline internal/repos/store.go:141]: This comment should mention org owned too I think

File: internal/repos/syncer.go [hunk 4]
```diff
@@ -641,33 +666,48 @@ func (s *Syncer) sync(ctx context.Context, svc *types.ExternalService, sourced *
...
 			if err != nil {
 				return Diff{}, errors.Wrap(err, "counting total user added repos")
 			}
 
-			userAdded, err := tx.CountUserAddedRepos(ctx, svc.NamespaceUserID)
+			// get either user ID or org ID. We cannot have both defined at the same time,
+			// so this naive addition should work
+			userAdded, err := tx.CountNamespacedRepos(ctx, svc.NamespaceUserID, svc.NamespaceOrgID)
 			if err != nil {
 				return Diff{}, errors.Wrap(err, "counting user added repos")
```
Comment (ryanslade) [inline internal/repos/syncer.go:679]: Error message should be updated.

**Review by ryanslade — APPROVED** [review]

File: internal/repos/syncer.go [hunk 5]
```diff
@@ -641,33 +666,48 @@ func (s *Syncer) sync(ctx context.Context, svc *types.ExternalService, sourced *
...
 			if siteAdded >= siteLimit || userAdded >= userLimit {
-				userStore := database.Users(tx.Handle().DB())
 				var username string
-				// We can ignore the error here as it's not fatal
-				if u, _ := userStore.GetByID(ctx, svc.NamespaceUserID); u != nil {
-					username = u.Username
+				var orgName string
+
+				if svc.NamespaceUserID != 0 {
+					userStore := database.Users(tx.Handle().DB())
```
Comment (tsenart) [inline internal/repos/syncer.go:689]: We're loading the org or user just to log its username in the logs. Can we skip that and only log their IDs?

**Review by tsenart — APPROVED** [review]

File: internal/repos/syncer.go (hunk 5 above)
Comment (kopancek) [reply internal/repos/syncer.go:689]: That's a valid point. I'll try to find out why it was added in the first place.

Comment (ryanslade) [reply internal/repos/syncer.go:689]: I think it was just to make debugging easier, happy if you just log the id instead.

Comment (kopancek) [reply internal/repos/syncer.go]: OK, removed that part of the code ✅

**Comment by tsenart** [comment]
Nice! Don't forget to deploy this manually, repo-updater only gets updated every other day.

**Pull request merged**